scripts/cmake/z_vcpkg_fixup_install_name_rpath.cmake

```
@@ -0,0 +1,411 @@
+# ├── lib
+# │   ├── libicudata.72.1.dylib
+# │   ├── libicudata.72.dylib -> libicudata.72.1.dylib
+# │   ├── libicudata.dylib -> libicudata.72.1.dylib
+# │   ├── ...
+# │   └── pkgconfig
+# │       ├── ...
+# │       └── icu-uc.pc
+# ├── share
+# │   └── icu
+# └── tools
+#     └── icu
+#         └── bin
+#             └── derb 
+# Here, it'wll do fix:
+# 1. for shared libraries in `lib` and `debug/lib`
+#    - fix its own install names to `@rpath` prefix 
+#    - fix its dependent shared library install name to `@rpath` prefix
+#       - include libraries only from the same port, assuming other libraries from other managed ports by vcpkg are already fixed!
+#       - exclude system libraries in `/usr/lib/` and `/System/Library`!
+#    - fix rpath to `@loader`(or using absolute path `CURRENT_INSTALLED_DIR`, like `/opt/vcpkg/installed/arm64-osx-dynamic/lib` or `/opt/vcpkg/installed/arm64-osx-dynamic/debug/lib`)
+# 2. for exe in `tools/{package}/bin/`
+#    - fix its dependent shared library install name to `@rpath` prefix
+#       - include libraries only from the same port in lib/, assuming other libraries from other managed ports by vcpkg are already fixed!
+#       - exclude system libraries in `/usr/lib/` and `/System/Library`!
+#    - fix rpath to `@loader/../../../`  to be relative with `lib/`
+function(z_vcpkg_fixup_install_name_rpath_in_dir)
+  find_program(
+    install_name_tool_cmd
+    NAMES install_name_tool
+    DOC "Absolute path of install_name_tool cmd"
+    REQUIRED
+  )
+
+  find_program(
+    otool_cmd
+    NAMES otool
+    DOC "Absolute path of otool cmd"
+    REQUIRED
+  )
+
+  message(DEBUG "Start fix install name and rpath for Mach-O files")
+
+  # Iterate over the folders. DON'T change order!
+  foreach(subfolder IN ITEMS lib debug/lib tools)
+    set(folder "${CURRENT_PACKAGES_DIR}/${subfolder}")
+
+    if(NOT EXISTS "${folder}")
+      continue()
+    endif()
+
+    get_filename_component(folder_name "${folder}" NAME)
+
+    # Get all files in the folder
+    file(GLOB_RECURSE files LIST_DIRECTORIES FALSE "${folder}/*")
+
+    set(fix_files)
+    # 1. Fix install name and rpath
+    foreach(file IN LISTS files)
+      # Check if the file is a Mach-O file
+      z_vcpkg_get_mach_o(type FILE "${file}")
+      if(type EQUAL 0)
+        continue()
+      endif()
+
+      # Skip if the file is a symlink
+      if(IS_SYMLINK "${file}")
+        continue()
+      endif()
+
+      message(DEBUG "Start to fix mach-O file install_name and rpath: ${file}")
+
+      # Get the directory of the file
+      get_filename_component(file_dir "${file}" DIRECTORY)
+      # Get the base name of the file
+      get_filename_component(filename "${file}" NAME)
+
+      # Get the major version
+      string(REGEX REPLACE "(lib[^\\.]+\\.[0-9]+).*\\.(dylib|so)" "\\1.\\2" filename_major "${filename}")
+      set(install_name_file "${file_dir}/${filename_major}")
+
+      # Try the major version file
+      if(NOT EXISTS "${install_name_file}")

```

It's not enough to test that the file exists; this will escape the symlink-only filter from line 268 and possibly change the ID of a non-symlinked file to a symlink. 

For a given `libfoo`, there are many possibilities:

- `libfoo.dylib`
- `libfoo.major.dylib`
- `libfoo.major.minor.dylib`
- `libfoo.major.minor.patch.dylib`

And usually, if more than two exist, all the others are symlinks to the lowest member in the above list, as on my install:

```shell
$ ls vcpkg_installed/arm64_osx/lib/libcom_err*
-rwxr-xr-x  2   53992 22 May 23:25 libcom_err.3.0.dylib
lrwxr-xr-x  1      20 22 May 23:25 libcom_err.3.dylib -> libcom_err.3.0.dylib
lrwxr-xr-x  1      20 22 May 23:25 libcom_err.dylib -> libcom_err.3.0.dylib
```

I propose instead:

```suggestion
      if(NOT EXISTS "${install_name_file}" OR IS_SYMLINK "${install_name_file}")
```

And see below for another change as well to account for `libfoo.major.minor.patch.dylib`.